docs/api-spec.md

```diff
@@ -0,0 +1,322 @@
+
+| Field Name | Field Type      | Requirement |
+|------------|-----------------|-------------|
+| `name`     | string          | REQUIRED    |
+| `value`    | `ArrayOrString` | REQUIRED    |
+
+### `ParamSpec`
+
+| Field Name    | Field Type | Requirement |
+|---------------|------------|-------------|
+| `name`        | string     | REQUIRED    |
+| `description` | string     | REQUIRED    |
+| `type`        | Enum: <br>- `"string"` (default) <br>- `"array"` | REQUIRED |
+| `default`     | `ArrayOrString` | REQUIRED |
+
+### `Step`
+
+| Field Name   | Field Type | Requirement |
+|--------------|------------|-------------|
+| `name`       | string     | REQUIRED    |
+| `image`      | string     | REQUIRED    |
+| `args`       | []string   | REQUIRED    |
+| `command`    | []string   | REQUIRED    |
+| `workingDir` | []string   | REQUIRED    |
+| `env`        | `[]EnvVar` | REQUIRED    |
+| `script`     | string     | REQUIRED    |
+
+**NB:** All other fields inherited from the [core.v1/Container](https://godoc.org/k8s.io/api/core/v1#Container) type supported by the Kubernetes implementation are **OPTIONAL** for the purposes of this spec.
+
+### `StepState`
+
+| Field Name    | Field Type                 | Requirement |
+|---------------|----------------------------|-------------|
+| `name`        | string                     | REQUIRED    |
+| `imageID`     | string                     | REQUIRED    |
+| `waiting`*    | `ContainerStateWaiting`    | REQUIRED    |
+| `running`*    | `ContainerStateRunning`    | REQUIRED    |
+| `terminated`* | `ContainerStateTerminated` | REQUIRED    |
+| `taskResults` | `TaskRunResult`            | REQUIRED    |
+| `taskSpec`    | `TaskSpec`                 | REQUIRED    |
+
+\* Only one of `waiting`, `running` or `terminated` can be returned at a time.
+
+### `TaskResult`
+
+| Field Name   | Field Type | Requirement |
+|--------------|------------|-------------|
+| `name`        | string    | REQUIRED    |
+| `description` | string    | REQUIRED    |
+
+### `TaskRunResult`
+
+| Field Name | Field Type | Requirement |
+|------------|------------|-------------|
+| `name`     | string     | REQUIRED    |
+| `value`    | string     | REQUIRED    |
+
+### `TaskSpec`
+
+| Field Name    | Field Type               | Requirement |
+|---------------|--------------------------|-------------|
+| `params`      | `[]ParamSpec`            | REQUIRED    |
+| `steps`       | `[]Step`                 | REQUIRED    |
+| `results`     | `[]TaskResult`           | REQUIRED    |
+| `workspaces`  | `[]WorkspaceDeclaration` | REQUIRED    |
+| `description` | string                   | REQUIRED    |
+
+### `WorkspaceBinding`
+
+| Field Name | Field Type   | Requirement |
+|------------|--------------|-------------|
+| `name`     | string       | REQUIRED    |
+| `emptyDir` | empty struct | REQUIRED    |
+
+**NB:** All other Workspace types supported by the Kubernetes implementation are **OPTIONAL** for the purposes of this spec.
```

Comment:
do we want to say anything more about this? in order to support tasks in a pipeline properly, the system would need to support _something_ besides emptyDir

Reply:
Yeah, this version of the doc is just to get a minimal "anything" spec out there, we can add more workspace types that platforms must support as we go.

It is _conceivable_ that a platform would only support emptyDir, though that would be a pretty serious limitation. So emptyDir should probably be the only REQUIRED option.